### PR TITLE
Don't crash if min and max are not set.

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/300/

**Comment by obino** [comment]
New OCD run after the extension to check all parameters not only min and max: https://ocd.appscale.com:8080/job/Daily%20Build/307/

**Comment by obino** [comment]
latest OCD run https://ocd.appscale.com:8080/job/Daily%20Build/319/

File: AppController/djinn.rb [hunk 1]
```diff
@@ -818,6 +855,53 @@ def set_parameters(djinn_locations, database_credentials, app_names, secret)
...
+      end
+
+      # Check that the value that came in is a String. If not, remove the
+      # parameter since we won't be able to translate it.
+      if val.class != String
+        begin
+          msg = "Removing parameter '" + key + "' with unknown value '" +\
+            val.to_s + "'."
+        rescue
+          msg = "Removing parameter '" + key + "' has unknown value."
```
Comment (menivaitsi) [inline AppController/djinn.rb]: "has" -> "with"

File: AppController/djinn.rb [hunk 2]
```diff
@@ -818,6 +855,53 @@ def set_parameters(djinn_locations, database_credentials, app_names, secret)
...
+        end
+        Djinn.log_warn(msg)
+        @options.delete(key)
+        next
+      end
+
+      # Let's check we can convert them now to the proper class.
```
Comment (menivaitsi) [inline AppController/djinn.rb]: "Let's check" -> "Let's check if"

File: AppController/djinn.rb [hunk 3]
```diff
@@ -818,6 +855,53 @@ def set_parameters(djinn_locations, database_credentials, app_names, secret)
...
+      if PARAMETERS_AND_CLASS[key] == Fixnum
+        begin
+          test_value = Integer(val)
+        rescue
+          msg = "Warning: parameter '" + key + "' is not an integer (" +\
+            val.to_s + "). Forcing it to 0."
+          Djinn.log_warn(msg)
+          @options[key] = 0.to_s
+        end
+      else
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Remove unnecessary code. You can leave the comment for clarity.

File: AppController/djinn.rb [hunk 4]
```diff
@@ -3017,8 +3099,8 @@ def wait_for_data()
   end
 
   def parse_options
-    if @options["appengine"]
-      @num_appengines = Integer(@options["appengine"])
+    if @options['appengine']
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Is this change for consistency?

Comment (obino) [inline AppController/djinn.rb]: Yes, we use ' everywhere else.

**Comment by obino** [comment]
tested with the configuration that created problems:

---
infrastructure : 'euca'
ips_layout:
  controller : node-1
machine : 'emi-9E05994D'
instance_type : 'm1.large'
zone : 'appscal1'
#min : 1
#max : 1

**Comment by obino** [comment]
Latest ocd run: https://ocd.appscale.com:8080/job/Daily%20Build/331/

**Comment by obino** [comment]
latest OCD run https://ocd.appscale.com:8080/job/Daily%20Build/343/

**Pull request merged**